src/core.rs

```diff
@@ -163,6 +180,803 @@ impl Resources {
         let output = String::from_utf8(csv_writer.into_inner()?)?;
         Ok(output)
     }
+    pub fn metrics(&self) -> Result<Registry, Box<dyn error::Error>> {
+        let mut registry = Registry::new();
+
+        for resource in &self.resources {
+            match resource {
+                Resource::Vm(vm) => {
+                    let osc_cost_version = vm
+                        .osc_cost_version
+                        .as_ref()
+                        .unwrap_or(&"v0.0.0".to_string())
```

Comment:
I don't think this is a good idea to compute with arbitrary values, if the value is not there, it should be a problem